dropwizard-jersey/src/main/java/io/dropwizard/jersey/params/AbstractParamConverterProvider.java

```diff
@@ -0,0 +1,60 @@
+package io.dropwizard.jersey.params;
+
+import com.google.common.base.Strings;
+import io.dropwizard.jersey.validation.JerseyParameterNameProvider;
+
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Type;
+
+public class AbstractParamConverterProvider implements ParamConverterProvider {
+
+    public AbstractParamConverterProvider() {
+    }
+
+    @Override
+    public <T> ParamConverter<T> getConverter(Class<T> rawType, Type genericType, Annotation[] annotations) {
+        if (AbstractParam.class.isAssignableFrom(rawType)) {
+            final String parameterName = JerseyParameterNameProvider.getParameterNameFromAnnotations(annotations).orElse("Parameter");
+            final Constructor<T> constructor;
+            try {
+                constructor = rawType.getConstructor(String.class, String.class);
+            } catch (NoSuchMethodException ignored) {
+                // The Param class did not have a (String, String) constructor. We return null,
+                // leaving Jersey to handle these parameters as it normally would.
+                return null;
+            }
+            return new ParamConverter<T>() {
```

Comment:
Possibly - the one thing we might not want to change is the null handling. Jerey's provider will throw an `IllegalArgumentException` if the value is null.  My change does not, as this would be a breaking change with previous versions of dropwizard.

Otherwise, we could change the exception handling to match abstract string provider or we could inherit from AbstractStringReader and just modify the behavior we don't like. Thoughts?